kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaChanges.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import io.delta.kernel.types.StructField;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * SchemaChanges encapsulates a list of added, removed, renamed, or updated fields in a schema
+ * change
+ *
+ * <p>ToDo: Possibly track moves/renames independently
```

Comment:
What is moves? Is this a reordering? is such a change included in `updatedFields`?

Reply:
Yeah reordering, they should be included in `updatedFields`. When it comes to validation, at some point we'll want checks that make sure we're not moving fields outside of their containing struct. I think that'll require a field ID to parent ID mapping that we can leverage to ensure that a given nested field ID is not moved to a field that is not its parent. That's separate though.